ckks/advanced/marshaler.go

```diff
@@ -5,84 +5,140 @@ import (
 	"math"
 )
 
+func (mParams *EncodingMatrixLiteral) MarshalBinarySize() (dataLen int) {
+	dataLen++                      // LinearTransform Type
+	dataLen++                      // LogN
+	dataLen++                      // LogSlots
+	dataLen++                      // LevelStart
+	dataLen++                      // #Levels
+	dataLen += len(mParams.Levels) // Levels
+	dataLen++                      // RepackImag2Real
+	dataLen += 8                   // Scaling
+	dataLen++                      // BitReversed
+	dataLen++                      // LogBSGSRatio
+	return
+}
+
 // MarshalBinary encode the target EncodingMatrixParameters on a slice of bytes.
 func (mParams *EncodingMatrixLiteral) MarshalBinary() (data []byte, err error) {
```

Comment:
I'd strongly advise against "optimal" binary marshalers for literal structs:
- they have to be adapted manually
- params serialization is usually not a performance bottleneck
- JSON serializers are usually much more user-friendly and composable
- `MarshalBinary()` can be `return mParams.MarshalJSON()`

Reply:
done